### PR TITLE
fix: aws cli issue  flatcar

### PR DESCRIPTION
**Pull request by faiq**
**What problem does this PR solve?**:

**Which issue(s) does this PR fix?**:

* https://jira.d2iq.com/browse/D2IQ-NUMBER

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: ansible/roles/providers/tasks/aws.yml [hunk 1]
```diff
@@ -19,11 +19,27 @@
...
+  pip:
+    name: "pyyaml==5.4.1"
+    executable: pip3
+    extra_args: "--no-build-isolation"
+  when:
+    - ansible_distribution == "Flatcar"
+
 - name: install aws clients
   pip:
     name: "awscli"
```
Comment (faiq) [inline ansible/roles/providers/tasks/aws.yml:40]: originally i tried pinning the version to 1.29.9 like we have in requirements-devkit.txt but this didn't work.

i then tried to see if i could use a newer pypy version https://www.pypy.org/download.html but this didn't work either.

File: ansible/roles/providers/tasks/aws.yml [hunk 2]
```diff
@@ -19,11 +19,27 @@
...
+  when:
+    - ansible_distribution == "Flatcar"
+
+- name: install pyyaml
+  pip:
+    name: "pyyaml==5.4.1"
+    executable: pip3
+    extra_args: "--no-build-isolation"
+  when:
+    - ansible_distribution == "Flatcar"
```
Comment (supershal) [inline ansible/roles/providers/tasks/aws.yml:36]: we should loop over using `with_items` in a single task.

File: ansible/roles/providers/tasks/aws.yml [hunk 3]
```diff
@@ -19,11 +19,27 @@
...
+    executable: pip3
+    extra_args: "wheel"
+  when:
+    - ansible_distribution == "Flatcar"
+
+- name: install pyyaml
+  pip:
+    name: "pyyaml==5.4.1"
+    executable: pip3
+    extra_args: "--no-build-isolation"
```
Comment (supershal) [inline ansible/roles/providers/tasks/aws.yml:34]: can you please brief documentation why extra_args `wheel` and `--no-build-isolation` for future reference.
the doc for `--no-build-isolation` is `Disable isolation when building a modern source`. but I am not sure what it means.

File: ansible/roles/providers/tasks/aws.yml (hunk 2 above)
Comment (faiq) [reply ansible/roles/providers/tasks/aws.yml:36]: They have different extra args so we can’t

Comment (faiq) [reply ansible/roles/providers/tasks/aws.yml:36]: Wait might have figured something out

File: ansible/roles/providers/tasks/aws.yml (hunk 3 above)
Comment (faiq) [reply ansible/roles/providers/tasks/aws.yml]: I'm not sure why, I copied this answer from stack overflow and tested it. 

I added a comment linking to the answer.

Comment (supershal) [reply ansible/roles/providers/tasks/aws.yml]: Thank you.

**Review by supershal — APPROVED** [review]
LGTM. lets wait for e2e tests to pass.

**Pull request merged**